pkg/gofr/datasource/redis/redis.go

```diff
@@ -1,23 +1,32 @@
-package datasource
+package redis
 
 import (
 	"context"
 	"fmt"
+	"time"
 
-	"github.com/redis/go-redis/extra/redisotel/v9"
-
+	otel "github.com/redis/go-redis/extra/redisotel/v9"
 	"github.com/redis/go-redis/v9"
+
+	"gofr.dev/pkg/gofr/datasource"
 )
 
-type RedisConfig struct {
+const redisPingTimeout = 5 * time.Second
+
+type Redis struct {
+	*redis.Client
+	logger datasource.Logger
+}
+
+type Config struct {
 	HostName string
 	Port     int
 	Options  *redis.Options
 }
 
-// newRedisClient return a redis client if connection is successful based on Config.
+// NewRedisClient return a redis client if connection is successful based on Config.
 // In case of error, it returns an error as second parameter.
-func NewRedisClient(config RedisConfig) (*redis.Client, error) {
+func NewRedisClient(config Config, logger datasource.Logger) (*Redis, error) {
```

Comment:
we can change the function name as well to `NewClient`, as its already in redis package so that would suffice.

Reply:
Sure will rename that to `NewClient`